sklearn/compose/tests/test_column_transformer.py

```diff
@@ -551,12 +552,15 @@ def test_column_transformer_remainder_numpy(key):
     assert_array_equal(ct.fit(X_array).transform(X_array), X_res_both)
 
 
-@pytest.mark.parametrize("key", [[0], slice(0, 1), np.array([True, False]),
-                                 ['first'], slice(None, 'first'),
-                                 slice('first', 'first')])
+@pytest.mark.parametrize(
+    "key", [[0], slice(0, 1), np.array([True, False]), ['first'], 'pd-index',
+            np.array(['first']), np.array(['first'], dtype=object),
+            slice(None, 'first'), slice('first', 'first')])
 def test_column_transformer_remainder_pandas(key):
     # test different ways that columns are specified with passthrough
     pd = pytest.importorskip('pandas')
+    if isinstance(key, six.string_types) and key == 'pd-index':
+        key = pd.Index(['first'])
```

Comment:
Yeah, it's a bit annoying, but I don't think so. I only do the skipping of pandas inside the test, meaning I can only call pandas methods inside the test (I think) 
(I can of course also simply do a loop over those possibilities inside the test instead of the parametrization, to avoid this extra check)